### ISSUE TITLE
Flash attention soft capping support

### ISSUE THREAD

**Issue by sparsh35** [google/jax#23300]
In Jax experimental pallas kernels for TPU , there is support for attn logits softcapping for paged attention but not for flash attention.
If support can be added for pallas flash kernels as well, as it can then be used in pytorch xla as well as vllm implementation.
Gemma 2 9b model works even with logit softcapping but 27 b doesn't. 

[PR for support of soft capping for Paged Attention](https://github.com/google/jax/commit/f0e36d5083457278339387285b06f170d28d87ac)

[Pytorch xla custom kernel integration for paged attention](https://github.com/pytorch/xla/pull/7704)

[Need for flash attention support for running Gemma 2 with VLLM on TPUs](https://github.com/vllm-project/vllm/issues/7950)

**Comment by sshkhr** [comment]
Is this still of interest @sharadmv @sparsh35 ?

Since no one is assigned yet, I would be interested in submitting a PR that adds softcap support to flash attention.

**Comment by sharadmv** [comment]
Yes feel free to contribute!

**Comment by sshkhr** [comment]
Hey @sharadmv I have a draft version of my patch ready. I want to double check something before I open a PR: the [JAX contributing guide](https://docs.jax.dev/en/latest/contributing.html) says "Pull requests typically comprise a single git commit". 

Since this is my first time contributing to JAX is it okay if I open a draft PR with my current changes (forward pass), so that I can get initial feedback (e.g. whether to the merge the soft capping and softmax scaling operations or not - similar to the [official FlashAttention codebase](https://github.com/Dao-AILab/flash-attention/blob/main/csrc/flash_attn/src/flash_fwd_kernel.h))

If needed I can squash my commits later on to meet the contributing guidelines. Does that work?

**Comment by sharadmv** [comment]
Yes that works

**Comment by sshkhr** [comment]
Cool, I opened a draft #28099, will mark it ready for review once I've received initial feedback and finished the backward pass
